enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,381 @@
+itself up in case it created any additional resources on the cluster or contributed any configuration.
+
+Therefore we plan to support turning a component on, but not turning it off.
+
+3. What are the implications for upgrades if a future upgrade would add a component or resource which would
+have been filtered out during install time?  
+
+There should be no implication here, the CVO has the list of annotations it will filter based on, if the
+new resources match those annotations, the new resources will also be filtered(never applied to the cluster).
+
+4. How prescriptive do we want to be about what can/can't be turned off?  Components need to opt into
+this by annotating their resources, so it's not completely arbitrary.
+
+This will need to be evaluated on a case by case basis as a component considers adding the annotation
+to its resources that will allow it to be filtered out/disabled.
+
+5. What if a user specifies a capability name to exclude that doesn't match anything?
+
+The installer won't know it's not going to match anything, so it should do nothing.  The CVO
+can tell it doesn't match anything, so it can provide a warning on the clusterversion resource, but
+it should not be an error since a user might put a keyword into the list in anticipation of upgrading
+to a new OCP version where the keyword will match (and thus exclude) a new capability, or perhaps
+the keyword used to match a capability that no longer exists after an upgrade.
+
+Along the same lines, if a user both "Excludes" and "Includes" a capability(if we ever add Include as
+a thing you can do), that should be treated as an error/invalid configuration.
+
+### Open Question
+
+1. What to do for components where disabling them has implications on other components or the way certain
+apis behave.  Example: disabling the internal registry changes the behavior of imagestreams
+(can't push to the imagestream anymore to push content to the internal registry) as well as the assumptions
+made by tools like new-app/new-build (create imagestreams that push to the internal registry).
+
+2. What to do(if anything) for components with interdependencies, to ensure a user doesn't break
+enabled components by disabling a dependency?  Options include:
+
+* Do nothing other than document dependencies so users know what not to turn off
+* Don't even annotate dependency resources for filtering, so if something is a dependency it cannot be turned off
+* Logic in the install or CVO that intelligently analyzes the filters the user has supplied and checks
+  for dependency issues (least desirable solution imho).
+
+3. Is Capabilities an ok name for these things/fields?  We'll want to get the naming right on the api to
+convey the right expectations for users.  We may revisit this name during implementation.
+
+### Test Plan
+
+1) Install clusters w/ the various add-on components included/excluded and confirm
+that the cluster is functional but only running the expected add-ons.
```

Comment:
As you mention [over here][1], possible combinations of optional components can be large.  I guess in several cases we can say "these should theoretically be orthogonal" and not bother with (much?) mixed testing.  But even with a handful of orthogonal components, that's still a 5x increase on top of all the other flavors we already have (single node?  FIPS?  Infra platform?  Proxy/mirror?  Etc., etc.).  Do we just pick a few of those, and hope the rest work out?  Is this the tipping point that moves us into some sort of randomized config sampling?  [Example from recently][2] where vSphere CI was randomly selecting a HW version, and that lead to a small amount of confusion while diagnosing the issue.  I don't think we need to invest a lot of additional energy on this in the enhancement itself, but if folks have ideas on where we'll go already, then recording those already-generated ideas seems reasonable.

[1]: https://github.com/openshift/enhancements/pull/922#discussion_r731217124
[2]: https://bugzilla.redhat.com/show_bug.cgi?id=2015187#c1